salt/daemons/test/test_saltkeep.py

```diff
@@ -232,20 +234,21 @@ def testBasic(self):
                 os.path.join('main', 'raet', 'main_master')))
         self.assertTrue(main.ha, ("0.0.0.0", raeting.RAET_PORT))
         self.assertIs(main.keep.auto, raeting.AutoMode.never.value)
-        self.assertDictEqual(main.keep.loadLocalData(), {'name': mainData['name'],
-                                                         'uid': 1,
-                                                         'ha': ['127.0.0.1', 7530],
-                                                         'iha': None,
-                                                         'natted': None,
-                                                         'fqdn': '1.0.0.127.in-addr.arpa',
-                                                         'dyned': None,
-                                                         'sid': 0,
-                                                         'puid': 1,
-                                                         'aha': ['0.0.0.0', 7530],
-                                                         'role': mainData['role'],
-                                                         'sighex': mainData['sighex'],
-                                                         'prihex': mainData['prihex'],
-                                                         })
+        self.assertDictEqual(main.keep.loadLocalData(),
+                             {'name': mainData['name'],
+                              'uid': 1,
+                              'ha': ['127.0.0.1', 7530],
+                              'iha': None,
+                              'natted': None,
+                              'fqdn': socket.getfqdn('127.0.0.1'),
+                              'dyned': None,
+                              'sid': 0,
+                              'puid': 1,
+                              'aha': ['0.0.0.0', 7530],
+                              'role': mainData['role'],
+                              'sighex': salt.utils.stringutils.to_str(mainData['sighex']),
+                              'prihex': salt.utils.stringutils.to_str(mainData['prihex']),
```

Comment:
Can you explain the use of `to_str` here (and elsewhere in this PR)? The `to_str` helper produces a `str` type on both Python 2 and 3. One common type of Python 3 incompatibility we've been cleaning up is when `to_str` is used instead of `to_bytes`. Since `str` and `bytes` are one and the same in Python 2, `to_str` will produce a `bytes` type on Python 2, but not on Python 3.

If what we need here is a `bytes` type, we need to be using the `to_bytes` helper instead.

Reply:
It's because with raet transport Salt uses Ioflo and Raet under the hood and these packages take no care about `unicode` and use `str` type by default. Here I've tried to work it around in salt to not expand all the great unicode work to raet.

Reply:
If exactly about these lines: raet generates these keys as a binary hex blobs, but after it packed and unpacked for network all the data becomes strings of `str` type that is binary in Py2 and unicode in Py3.

Reply:
OK, thanks for the explanation, this is good to merge then. I just wanted to make sure this value wasn't intended to be a bytes type, because this would break it on Python 3 if that were the case.

Reply:
Thank you for pointing on this.